packages/headless/src/api/analytics/insight-analytics.test.ts

```diff
@@ -0,0 +1,120 @@
+import {CoveoAnalyticsClient} from 'coveo.analytics';
+import pino from 'pino';
+import {getConfigurationInitialState} from '../../features/configuration/configuration-state';
+import {buildMockFacetRequest} from '../../test/mock-facet-request';
+import {buildMockFacetResponse} from '../../test/mock-facet-response';
+import {buildMockFacetValue} from '../../test/mock-facet-value';
+import {buildMockInsightState} from '../../test/mock-insight-state';
+import {buildMockQueryState} from '../../test/mock-query-state';
+import {buildMockResult} from '../../test/mock-result';
+import {buildMockSearchState} from '../../test/mock-search-state';
+import {
+  configureInsightAnalytics,
+  InsightAnalyticsProvider,
+  StateNeededByInsightAnalyticsProvider,
+} from './insight-analytics';
+
+describe('insight analytics', () => {
+  const logger = pino({level: 'silent'});
+
+  it('should be enabled by default', () => {
+    const state = buildMockInsightState();
+    expect(
+      configureInsightAnalytics({state, logger}).coveoAnalyticsClient instanceof
+        CoveoAnalyticsClient
+    ).toBe(true);
```

Comment:
in the case when the client is disabled it will return `NoopAnalytics` 
in coveo.analytics.js:
![image](https://user-images.githubusercontent.com/85293211/177360109-10c8a417-f4fb-47ff-8a61-ac055e571803.png)